src/app/clusters/operational-state-server/operational-state-server.h

```diff
@@ -0,0 +1,210 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include "OperationalStateDataProvider.h"
+
+#include <app/AttributeAccessInterface.h>
+#include <app/CommandHandlerInterface.h>
+#include <app/clusters/operational-state-server/operational-state-delegate.h>
+#include <app/util/af.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace OperationalState {
+
+class Uncopyable
+{
+protected:
+    Uncopyable() {}
+    ~Uncopyable() {}
+
+private:
+    Uncopyable(const Uncopyable &);
+    Uncopyable & operator=(const Uncopyable &);
+};
+
+class OperationalStateServer : public CommandHandlerInterface, public AttributeAccessInterface, public Uncopyable
+{
+
+public:
+    template <typename T>
+    using DataModelListTemplate          = app::DataModel::List<T>;
+    using OperationalStateStructType     = app::Clusters::OperationalState::Structs::OperationalStateStruct::Type;
+    using OperationalStateStructTypeList = app::DataModel::List<OperationalStateStructType>;
+    using PhaseList                      = chip::app::DataModel::List<const chip::CharSpan>;
+    /**
+     * Init the operational state server.
+     * @param void
+     * @return CHIP_ERROR CHIP_NO_ERROR on success, or corresponding error code.
+     */
+    CHIP_ERROR Init();
+
+    // Inherited from CommandHandlerInterface
+    void InvokeCommand(HandlerContext & ctx) override;
+    CHIP_ERROR EnumerateAcceptedCommands(const ConcreteClusterPath & cluster, CommandIdCallback callback, void * context) override;
+
+    /// IM-level implementation of read
+    ///
+    /// Returns appropriately mapped CHIP_ERROR if applicable (may return CHIP_IM_GLOBAL_STATUS errors)
+    CHIP_ERROR Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder) override;
+
+    /**
+     * Set operational state list.
+     * @param operationalStateList The operational state list for which to save.
+     *  Template for save operational state alias cluster
+     * @return CHIP_ERROR CHIP_NO_ERROR on success, or corresponding error code.
+     */
+    template <typename T>
+    CHIP_ERROR SetOperationalStateList(const DataModelListTemplate<T> & operationalStateList);
+
+    /**
+     * Get operational state list.
+     * @param operationalStateList The pointer to save operational state list.
```

Comment:
What is being saved where?  Please document which params are input and which are output.

Reply:
Done